File: app/src/main/java/com/akameko/testforupstarts/repository/room/JeansDao.java
package com.akameko.testforupstarts.repository.room;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.akameko.testforupstarts.repository.pojos.Jeans;

import java.util.List;

@Dao
public interface JeansDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<Jeans> jeanses);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Jeans jeans);

    @Delete
    void delete(Jeans jeans);

    @Query("DELETE FROM jeans")
    void deleteAll();

    @Query("SELECT * FROM jeans")
    List<Jeans> getAllItems();

    @Query("SELECT * FROM jeans WHERE id == :id")
    Jeans getItemById(Integer id);
}

File: app/src/main/java/com/akameko/testforupstarts/repository/room/JeansDao.kt
package com.akameko.testforupstarts.repository.room

import androidx.room.*
import com.akameko.testforupstarts.repository.pojos.Jeans

@Dao
interface JeansDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(jeanses: List<Jeans?>?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(jeans: Jeans?)

    @Delete
    fun delete(jeans: Jeans?)

    @Query("DELETE FROM jeans")
    fun deleteAll()

    @get:Query("SELECT * FROM jeans")
    val allItems: List<Jeans?>?

    @Query("SELECT * FROM jeans WHERE id == :id")
    fun getItemById(id: Int?): Jeans?
}